core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/io/FastDoubleParser.kt

```diff
@@ -0,0 +1,233 @@
+            buildSet {
+                val getChar = this@fromLocalWithFallBack
+                val char = getChar(localDecimalFormatSymbols).lowercaseChar()
+                add(char)
+
+                // add fallback char if it's safe to do so
+                val fallbackChar = getChar(fallbackDecimalFormatSymbols).lowercaseChar()
+                if (fallbackChar !in localChars && !localStrings.any { fallbackChar in it }) {
+                    add(fallbackChar)
+                }
+
+                // Fixes NBSP and other whitespace characters not being recognized if the user writes space instead.
+                if (char.isWhitespace()) add(' ')
+
+                // add additional chars if needed
+                for (additional in additionals) {
+                    val lowercase = additional.lowercaseChar()
+                    if (lowercase !in localChars && !localStrings.any { lowercase in it }) {
+                        add(lowercase)
+                    }
+                }
+            }
+
+        /**
+         * Builds a set with the specified string from [localDecimalFormatSymbols] and
+         * its fallback string from [fallbackDecimalFormatSymbols] if it's safe to do so.
+         * [additionals] will be added to the set too, when they're safe to add.
+         */
+        fun ((DecimalFormatSymbols) -> String).fromLocalWithFallBack(vararg additionals: String): Set<String> =
+            buildSet {
+                val getString = this@fromLocalWithFallBack
+                val string = getString(localDecimalFormatSymbols).lowercase()
+                add(string)
+
+                // add fallback string if it's safe to do so
+                val fallbackString = getString(fallbackDecimalFormatSymbols).lowercase()
+                if (!fallbackString.any { it in localChars } && fallbackString !in localStrings) {
+                    add(fallbackString)
+                }
+
+                // Fixes NBSP and other whitespace characters not being recognized if the user writes space instead.
+                if (string.isBlank()) add(" ")
+
+                // add additional strings if needed
+                for (additional in additionals) {
+                    val lowercase = additional.lowercase()
+                    if (!lowercase.any { it in localChars } && lowercase !in localStrings) {
+                        add(lowercase)
+                    }
+                }
+            }
+
+        return NumberFormatSymbols.fromDecimalFormatSymbols(localDecimalFormatSymbols)
+            .withPlusSign(setOf('+'))
+            .withDecimalSeparator(DecimalFormatSymbols::getDecimalSeparator.fromLocalWithFallBack())
+            .withGroupingSeparator(DecimalFormatSymbols::getGroupingSeparator.fromLocalWithFallBack())
+            .withExponentSeparator(DecimalFormatSymbols::getExponentSeparator.fromLocalWithFallBack())
```

Comment:
What does it mean `set(get.fromLocalWithFallBack)` construction?

Reply:
you can read it as a sentence :) `::getDecimalSeparator.fromLocalWithFallBack()`. This means it calls `localDecimalFormatSymbols.getDecimalSeparator()` first and then adds fallback characters from `fallbackDecimalFormatSymbols.getDecimalSeparator()`.

Reply:
I made it work on the function reference so that the function can be called from different `DecimalFormatSymbols` instances